201-dscforlinux-extension-on-ubuntu/azuredeploy.json

```diff
@@ -0,0 +1,297 @@
+            "metadata": {
+                "description": "Password for the Virtual Machine."
+            }
+        },
+        "location": {
+            "type": "string",
+            "metadata": {
+                "description": "The location where the Virtual Machine will be deployed."
+            },
+            "allowedValues": [
+                "Australia East",
+                "Australia Southeast",
+                "Brazil South",
+                "Central US",
+                "East Asia",
+                "East US",
+                "East US 2",
+                "Japan East",
+                "Japan West",
+                "North Central US",
+                "North Europe",
+                "South Central US",
+                "Southeast Asia",
+                "West Europe",
+                "West US"
+            ]
+        },
+        "newStorageAccountName": {
+            "type": "string",
+            "metadata": {
+                "description": "Unique DNS Name for the Storage Account where the Virtual Machine's disks will be placed."
+            }
+        },
+        "vmName": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of the vm, will be used as DNS Name for the Public IP used to access the Virtual Machine."
+            }
+        },
+        "vmSize": {
+            "type": "string",
+            "metadata": {
+                "description": "VM size"
+            },
+            "allowedValues": [
+                "Standard_D1",
+                "Standard_D2",
+                "Standard_D3",
+                "Standard_D4",
+                "Standard_D11",
+                "Standard_D12",
+                "Standard_D13",
+                "Standard_D14"
+            ]
+        },
+        "ubuntuOSVersion": {
+            "type": "string",
+            "defaultValue": "14.04.3-LTS",
+            "metadata": {
+                "description": "The Ubuntu version"
+            },
+            "allowedValues": [
+                "14.04.3-LTS",
+                "12.04.5-LTS"
+            ]
+        },
+        "Mode": {
+            "type": "string",
+            "defaultValue": "Push",
+            "metadata": {
+                "description": "The functional mode, push MOF configuration (Push), distribute MOF configuration (Pull), install custom DSC module (Install)"
+            },
+            "allowedValues": [
+                "Push",
+                "Pull",
+                "Install"
+            ]
+        },
+        "MofFileUri": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "The URI of the public MOF file"
+            }
+        },
+        "ResourceZipFileUri": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "The URI of the public custom resource ZIP file to install"
+            }
```

Comment:
You have the MofFileUri and ResourceZipFileUri at this point in the template. Is there a way to avoid having the user enter 5 more parameters below?